news/6069.bugfix.rst

```diff
@@ -0,0 +1,3 @@
+Fix a bug in implicit metadata collection where, on seeing
+``pkg_resources.require("foo")``, PyInstaller would only collect ``foo``\ s
+metadata instead of ``foo`` plus all its dependencies.
```

Comment:
I think "... instead of metadata of ``foo`` and all its dependencies." would make it a bit clearer that this is only about metadata, as opposed to *whole* ``foo`` and its dependencies.